`test__get_vector_similarity_query_support_true` and `test__get_vector_similarity_query_support_false` for `OpenSearchDocumentStore` are flaky

**Describe the bug**
In this [test run](https://github.com/deepset-ai/haystack/runs/7947329329?check_suite_focus=true) on the main branch, the tests `test__get_vector_similarity_query_support_true` and `test__get_vector_similarity_query_support_false` failed due to an `AssertionError`. This error didn't occur before.

**Error message**
```
E       AssertionError: assert {'script_scor...'knn_score'}}} == {'script_scor...'knn_score'}}}
E         Differing items:
E         {'script_score': {'query': {'match_all': {}}, 'script': {'lang': 'knn', 'params': {'field': 'FooField', 'query_value': [[0.0, 8.4368011761e-313], [1.3498881881773e-311, nan]], 'space_type': 'innerproduct'}, 'source': 'knn_score'}}} != {'script_score': {'query': {'match_all': {}}, 'script': {'lang': 'knn', 'params': {'field': 'FooField', 'query_value': [[0.0, 8.4368011761e-313], [1.3498881881773e-311, nan]], 'space_type': 'innerproduct'}, 'source': 'knn_score'}}}
```

**Additional context**
This error most likely is due to the fact that we create a random numpy array [here](https://github.com/deepset-ai/haystack/blob/d5e36ce6b4b61a1237a6b5c76d0571b9aef01de2/test/document_stores/test_opensearch.py#L27) using `np.ndarray`. As stated in the [numpy documentation](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html), this array might contain `nan` values. And as stated in this section of the [numpy documentation](https://numpy.org/doc/stable/user/misc.html), comparing two `nan` values for equality will always have `False` as a result.

We should make sure that the randomly initialized numpy array does not contain any `nan` values, for example by using `np.random.random_sample`.

--- a/test/document_stores/test_opensearch.py
+++ b/test/document_stores/test_opensearch.py
@@ -24,7 +24,7 @@ class TestOpenSearchDocumentStore:
 
     # Constants
 
-    query_emb = np.ndarray(shape=(2, 2), dtype=float)
+    query_emb = np.random.random_sample(size=(2, 2))
     index_name = "myindex"
 
     # Fixtures

